netket/operator/_abstract_operator.py

```diff
@@ -64,7 +67,6 @@ def T(self) -> "AbstractOperator":
     @abc.abstractmethod
     def dtype(self) -> DType:
         """The dtype of the operator's matrix elements ⟨σ|Ô|σ'⟩."""
-        raise NotImplementedError
```

Comment:
this method is already abstract so will never be called.
There's no need to add this raise and only pollutes CI coverage reports